test/unit/asrouter/ASRouter.test.js

```diff
@@ -1053,8 +1053,19 @@ describe("ASRouter", () => {
         await Router.onMessage(msg);
 
         assert.calledOnce(MessageLoaderUtils.installAddonFromURL);
-        assert.calledWithExactly(MessageLoaderUtils.installAddonFromURL, msg.target.browser, "foo.com");
+        assert.calledWithExactly(MessageLoaderUtils.installAddonFromURL, msg.target.browser, "foo.com", undefined);
```

Comment:
@piatra (if I recall correctly) the main reason why I chose to add `undefined` to this assertion was to verify (as part of the automated unit tests) that by default `installAddonFromURL` is called with that parameter passed as `undefined` (so that the default "amo" source is going to be used as expected in those cases)

Reply:
I see. Then in order to test that the default parameter is correctly set I would `spy` on `MessageLoaderUtils.installAddonFromURL` and check that `AddonManager.getInstallForURL` is called with the correct (default) telemetry source value.
 
My nit here is that `calledWithExactly(..., undefined)` isn't a very useful information.

Reply:
@piatra I took another look and:

- the scenario with an explicit `telemetrySource` is already covered by the new test case added in the same file (`"should call installAddonFrom with correct arguments when a custom telemetrySource is included"` defined in `ASRouter.test.js`)

- testing that the default install `telemetrySource` is `"amo"` should be already covered by the `"should call the Addons API when passed a valid URL"` test case included in `MessageLoaderUtils.test.js`.

- in this existing test case I had to add `undefined` because otherwise the `calledWithExactly` assertion fails (because `calledWithExactly` expects that all the parameters are going to be explicitly asserted, even the `undefined` ones).

Reply:
@piatra on a second thought, I can merge "should call installAddonFrom with correct arguments when a custom telemetrySource is included" into this test case to avoid the `calledWithExactly(..., undefined)` assertion, how that sounds to you?

Reply:
Sounds good to me, thanks!

Reply:
done (and thanks to you for your patience ;-))